sim/hunter/items.go

```diff
@@ -355,10 +358,146 @@ func init() {
+					aura.RemoveStack(sim)
+				}
+			},
+		})
+
+		spell := character.GetOrRegisterSpell(core.SpellConfig{
+			ActionID: core.ActionID{SpellID: 468388},
+			Flags:    core.SpellFlagNoOnCastComplete | core.SpellFlagOffensiveEquipment,
+
+			Cast: core.CastConfig{
+				CD: core.Cooldown{
+					Timer:    character.NewTimer(),
+					Duration: time.Minute * 2,
+				},
+			},
+			ApplyEffects: func(sim *core.Simulation, target *core.Unit, spell *core.Spell) {
+				lockedIn.Activate(sim)
+				lockedIn.SetStacks(sim, 3)
+			},
+		})
+
+		character.AddMajorCooldown(core.MajorCooldown{
+			Spell: spell,
+			Type:  core.CooldownTypeDPS,
+		})
+	})
+
+	// https://www.wowhead.com/classic/item=230237/arcane-infused-gem
+	core.NewItemEffect(ArcaneInfusedGem, func(agent core.Agent) {
+		character := agent.GetCharacter()
+
+		arcaneDetonation := character.RegisterSpell(core.SpellConfig{
+			ActionID: 		core.ActionID{SpellID: 467447},
+			SpellSchool: 	core.SpellSchoolArcane,
+			DefenseType: 	core.DefenseTypeMagic,
+			ProcMask: 		core.ProcMaskSpellDamage,
+			Flags: 			core.SpellFlagNoOnCastComplete | core.SpellFlagPassiveSpell,
+
+			DamageMultiplier: 1,
+			ThreatMultiplier: 1,
+			BonusCoefficient: 0,
+
+			ApplyEffects: func(sim *core.Simulation, target *core.Unit, spell *core.Spell) {
+				for _, aoeTarget := range sim.Encounter.TargetUnits {
+					damage := sim.Roll(185, 210)
+					spell.CalcAndDealDamage(sim, aoeTarget, damage, spell.OutcomeMagicCrit)
+				}
+			},
+		})
+
+		arcaneInfused := character.RegisterAura(core.Aura{
+			Label: "Arcane Infused",
+			ActionID: core.ActionID{SpellID: 467446},
+			Duration: time.Second * 15,
+			OnCastComplete: func(aura *core.Aura, sim *core.Simulation, spell *core.Spell) {
+				maxTargetsPerCast := int32(5)
+				// Uses same targeting code as multi-shot however the detonations occur at cast time rather than when the shots land
+				if spell.SpellCode == SpellCode_HunterMultiShot {
+					curTarget := sim.Environment.Encounter.TargetUnits[0]
+					for hitIndex := int32(0); hitIndex < min(3, character.Env.GetNumTargets()); hitIndex++ {
+						arcaneDetonation.Cast(sim, curTarget)
+						curTarget = sim.Environment.NextTargetUnit(curTarget)
+					}
+				}
+				// 1 explosion per target up to 5 targets per carve cast
+				if spell.SpellCode == SpellCode_HunterCarve {
+					curTarget := sim.Environment.Encounter.TargetUnits[0]
+					for hitIndex := int32(0); hitIndex < min(maxTargetsPerCast, character.Env.GetNumTargets()); hitIndex++ {
+						arcaneDetonation.Cast(sim, curTarget)
+						curTarget = sim.Environment.NextTargetUnit(curTarget)
+					}
+				}
+			},
+		})
+
+		spell := character.GetOrRegisterSpell(core.SpellConfig{
+			ActionID: core.ActionID{SpellID: arcaneInfused.ActionID.SpellID},
+			Flags:    core.SpellFlagNoOnCastComplete | core.SpellFlagOffensiveEquipment,
+
+			Cast: core.CastConfig{
+				CD: core.Cooldown{
+					Timer:    character.NewTimer(),
+					Duration: time.Second * 90,
+				},
```

Comment:
Looks like https://www.wowhead.com/classic/item=230237/arcane-infused-gem uses the shared offensive trinket cooldown